### ISSUE TITLE
Android - No sound on notifications?

### ISSUE THREAD

**Issue by cristianoccazinsp** [zo0r/react-native-push-notification#1432]
# Bug

Using `{playSound: true, soundName: 'default'}` doesn't work with `localNotification`.  This affects both iOS and Android. Are there limitations on local notifications? I am able to turn on sound on Android if I configure the notification channel (from the OS settings) to use sound.

## Environment info

Library version: 3.4.0

**Comment by UxmanKaxmi** [comment]
okay so this might sound a little stupid but it might be an issue for you too.

I was using the whole string "android.resource://com.xyz/raw/my_sound" instead of just using "my_sound".  :D 
You just have to use the exact name, not the whole string. I think someone should look at the comment which is written in the latest commit.

**Comment by cristianoccazinsp** [comment]
I'm using just `'default'` (no custom sounds), yet the notification is set as no sound. What version are you using?

Probably related:
https://github.com/zo0r/react-native-push-notification/issues/1143
https://github.com/zo0r/react-native-push-notification/issues/1045

**Comment by UxmanKaxmi** [comment]
Oh. My default sound was playing fine.  My version is
`    "react-native-push-notification": "^3.4.0",
`

`    "react-native": "0.62.2",
`

and I am testing on my Samsung Note 8.

**Comment by cristianoccazinsp** [comment]
I've tested on a Pixel 2 (Android 10) and Motorola G5 (Android 8.1), neither seem to work.  It seems like the notification channel is created muted as default.

**Comment by UxmanKaxmi** [comment]
Can you show the code?

**Comment by cristianoccazinsp** [comment]
Sample Code:

```
import PushNotification from 'react-native-push-notification';


PushNotification.configure({

  // (optional) Called when Token is generated (iOS and Android)
  // NOT using push notifications from the server right now, this step is not done.
  onRegister: function(token) {
  },

  // (required) Called when a remote or local notification is opened or received
  onNotification: function(notification) {

    // required on iOS only
    if(Platform.OS == 'ios'){
      notification.finish(PushNotificationIOS.FetchResult.NewData);
    }
  },


  // ANDROID ONLY: GCM or FCM Sender ID (product_number) (optional - not required for local notifications, but is need to receive remote push notifications)
  //senderID: "YOUR GCM (OR FCM) SENDER ID",

  // IOS ONLY (optional): default: all - Permissions to register.
  permissions: {
    alert: true,
    badge: true,
    sound: true
  },

  // Should the initial notification be popped automatically
  // default: true
  popInitialNotification: true,

  /**
   * (optional) default: true
   * - Specified if permissions (ios) and token (android and ios) will requested or not,
   * - if not, you must call PushNotificationsHandler.requestPermissions() later
   * - if you are not using remote notification or do not have Firebase installed, use this:
   *     requestPermissions: Platform.OS === 'ios'
   */
  requestPermissions: Platform.OS === 'ios',
});



function showNotification(title, message, id, vibrate, sound, ongoing=false){
  PushNotification.localNotification({
    /* Android Only Properties */
    id: id,
    autoCancel: true,
    vibrate: vibrate,
    vibration: vibrate ? 300 : undefined,
    priority: "high",
    visibility: "public",
    importance: "high",
    ongoing: ongoing,

    /* iOS only properties */
    //alertAction: 'view',
    userInfo: {id: id}, // required for ios local notification

    /* iOS and Android properties */
    title: title,
    message: message, // (required)
    playSound: sound,
    soundName: sound ? 'default' : undefined,
    // number: number // silly library, iOS requires number, while android string...

  });
}


showNotification("Test", `Test.`, '0', true, true);
```

**Comment by Gabsys** [comment]
Check in ur Settings -> Notification -> Your App -> Sound, is a sound chosen?

**Comment by Dallas62** [comment]
Hum...
I think the problem comes from the management of sounds:

https://github.com/zo0r/react-native-push-notification/blob/master/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java#L363

Can you try to change in your node_modules, the file mentioned and change:
```java
            if (!bundle.containsKey("playSound") || bundle.getBoolean("playSound")) {
                soundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
                
                String soundName = bundle.getString("soundName");
                
                if (soundName != null) {
                    if (!"default".equalsIgnoreCase(soundName)) {

                        // sound name can be full filename, or just the resource name.
                        // So the strings 'my_sound.mp3' AND 'my_sound' are accepted
                        // The reason is to make the iOS and android javascript interfaces compatible

                        int resId;
                        if (context.getResources().getIdentifier(soundName, "raw", context.getPackageName()) != 0) {
                            resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
                        } else {
                            soundName = soundName.substring(0, soundName.lastIndexOf('.'));
                            resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
                        }

                        soundUri = Uri.parse("android.resource://" + context.getPackageName() + "/" + resId);

                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) { // API 26 and higher
                            channel_id = channel_id + "-" + soundName;
                        }
                    }
                }

                notification.setSound(soundUri);
            }
```

By:

```java
            if (!bundle.containsKey("playSound") || bundle.getBoolean("playSound")) {
                soundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
                
                String soundName = bundle.getString("soundName");
                
                if (soundName != null) {
                    if (!"default".equalsIgnoreCase(soundName)) {

                        // sound name can be full filename, or just the resource name.
                        // So the strings 'my_sound.mp3' AND 'my_sound' are accepted
                        // The reason is to make the iOS and android javascript interfaces compatible

                        int resId;
                        if (context.getResources().getIdentifier(soundName, "raw", context.getPackageName()) != 0) {
                            resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
                        } else {
                            soundName = soundName.substring(0, soundName.lastIndexOf('.'));
                            resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
                        }

                        soundUri = Uri.parse("android.resource://" + context.getPackageName() + "/" + resId);
                    }
                } else {
                    soundName = "default";
                }

                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) { // API 26 and higher
                    channel_id = channel_id + "-" + soundName;
                }

                notification.setSound(soundUri);
            }
```

**Comment by Gabsys** [comment]
> ```java
> soundUri
> ```

Didn't work for me. The Sound selected by default is still "None" instead of the default sound

**Comment by Gabsys** [comment]
> Hum...
> I think the problem comes from the management of sounds:
> 
> https://github.com/zo0r/react-native-push-notification/blob/master/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java#L363
> 
> Can you try to change in your node_modules, the file mentioned and change:
> 
> ```java
>             if (!bundle.containsKey("playSound") || bundle.getBoolean("playSound")) {
>                 soundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
>                 
>                 String soundName = bundle.getString("soundName");
>                 
>                 if (soundName != null) {
>                     if (!"default".equalsIgnoreCase(soundName)) {
> 
>                         // sound name can be full filename, or just the resource name.
>                         // So the strings 'my_sound.mp3' AND 'my_sound' are accepted
>                         // The reason is to make the iOS and android javascript interfaces compatible
> 
>                         int resId;
>                         if (context.getResources().getIdentifier(soundName, "raw", context.getPackageName()) != 0) {
>                             resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
>                         } else {
>                             soundName = soundName.substring(0, soundName.lastIndexOf('.'));
>                             resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
>                         }
> 
>                         soundUri = Uri.parse("android.resource://" + context.getPackageName() + "/" + resId);
> 
>                         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) { // API 26 and higher
>                             channel_id = channel_id + "-" + soundName;
>                         }
>                     }
>                 }
> 
>                 notification.setSound(soundUri);
>             }
> ```
> 
> By:
> 
> ```java
>             if (!bundle.containsKey("playSound") || bundle.getBoolean("playSound")) {
>                 soundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
>                 
>                 String soundName = bundle.getString("soundName");
>                 
>                 if (soundName != null) {
>                     if (!"default".equalsIgnoreCase(soundName)) {
> 
>                         // sound name can be full filename, or just the resource name.
>                         // So the strings 'my_sound.mp3' AND 'my_sound' are accepted
>                         // The reason is to make the iOS and android javascript interfaces compatible
> 
>                         int resId;
>                         if (context.getResources().getIdentifier(soundName, "raw", context.getPackageName()) != 0) {
>                             resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
>                         } else {
>                             soundName = soundName.substring(0, soundName.lastIndexOf('.'));
>                             resId = context.getResources().getIdentifier(soundName, "raw", context.getPackageName());
>                         }
> 
>                         soundUri = Uri.parse("android.resource://" + context.getPackageName() + "/" + resId);
>                     }
>                 } else {
>                     soundName = "default";
>                 }
> 
>                 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) { // API 26 and higher
>                     channel_id = channel_id + "-" + soundName;
>                 }
> 
>                 notification.setSound(soundUri);
>             }
> ```

Didn't work for me. The Sound selected by default is still "None" instead of the default sound

**Comment by Dallas62** [comment]
@Gabsys 
This fix is not for remote notification since remote notification use a default channel which is not impacted by this change, will make the change for both ASAP.

**Comment by cristianoccazinsp** [comment]
> Check in ur Settings -> Notification -> Your App -> Sound, is a sound chosen?

Sound is available, but no chosen ("no sound" / "none"). Of course I could choose a sound there, and then it works, but it should be turned on by default.

**Comment by cristianoccazinsp** [comment]
@Dallas62  what about iOS **local** notifications? They seem muted as well,

**Comment by Gabsys** [comment]
> > Check in ur Settings -> Notification -> Your App -> Sound, is a sound chosen?
> 
> Sound is available, but no chosen ("no sound" / "none"). Of course I could choose a sound there, and then it works, but it should be turned on by default.

Yes, I face the same issue. I believe @Dallas62 is fixing it now

**Comment by Dallas62** [comment]
Hi,
I put some changes in `dev` branch, can you test them ?
```
yarn add zo0r/react-native-push-notification#dev
```
OR
```
npm install zo0r/react-native-push-notification#dev
```

**Comment by Dallas62** [comment]
You probably need to reinstall the application, but if this is already released, you can also set:
```xml
<meta-data
    android:name="com.google.firebase.messaging.default_notification_channel_id"
    android:value="rn-push-notification-channel-id-default-4-300" />
```
in AndroidManifest

**Comment by anishsrinivasan** [comment]
> You probably need to reinstall the application, but if this is already released, you can also set:
> 
> ```
> <meta-data
>     android:name="com.google.firebase.messaging.default_notification_channel_id"
>     android:value="rn-push-notification-channel-id-default-4-300" />
> ```
> 
> in AndroidManifest

Hi @Dallas62 ,
Local Push Notifications is working properly with Custom Sound, but when I send a notification from Firebase (FCM) with custom sound, it's not working properly. (Default Notification Sound is playing instead)

My bet, It could be something to do with the payload sent through the FCM as well.

The Payload which I'm trying to send
>    const payload = {
>     notification: {
>        alert: "You have a notification",
>        title: "This is a Notification",
>        body: "This is the body of the notification message.",
>        sound: "sound.mp3",
>        soundName: "sound.mp3",
>        sticky: "true",
>        image:
>          "https://www.gstatic.com/devrel->devsite/prod/v84899ba5ac366dd19b845bb4579ea9262ac5ac73d5db61a8fa440a5f2fc65a26/fi>rebase/images/lockup.png",
>      },
>      data: {},
>   };` 
>

On Foreground 

When I console notification of onNotification(notification) 

>
>color: null
>data: {}
>finish: ƒ finish()
>foreground: true
>id: "-1628629521"
>message: "This is the body of the notification message."
>sound: "sound"
>title: "This is a Notification"
>userInteraction: false
>__proto__: Object
>

As you see the sound:"sound.mp3" is received as "sound", Is that could be an issue? 



I also tried installing the dev & set the above lines in AndroidManifest, No result.

**Comment by Dallas62** [comment]
Hi @anishsrinivasan 
This is not the same issue, but I think this is due to the Channel of Android which is configured for a specific sound.
Since the library as a poor support of channels, in the current state of the library, I don't have a workaround...

**Comment by cristianoccazinsp** [comment]
> You probably need to reinstall the application, but if this is already released, you can also set:
> 
> ```
> <meta-data
>     android:name="com.google.firebase.messaging.default_notification_channel_id"
>     android:value="rn-push-notification-channel-id-default-4-300" />
> ```
> 
> in AndroidManifest

Do we need this even if we only use local notifications? I will try to test it once I get time.

**Comment by Dallas62** [comment]
This is not needed for local notification, only remote.
But if you can try this:
https://github.com/zo0r/react-native-push-notification/issues/1432#issuecomment-627774916

**Comment by hubermat** [comment]
The #dev version solves the problem for local notifications with default sound for me. Thank you very much! When can we expect a new version with this fix?

**Comment by Dallas62** [comment]
Hi @hubermat
Thanks for the feedback,
I will check this in the next tow days

**Comment by wbercx** [comment]
Sounds seem to be fixed in the latest release, but now they somehow defeat the volume settings of the OS.

I've got my notification volume silenced and the notifications for this app come out at full volume anyway. Some of our users reported it's really soft for them, despite having their phones set at full volume.

**Comment by Dallas62** [comment]
Hi @wbercx 
Is it for remote notification ?
If yes, set the default channel of firebase in the AndroidManifest:
```xml
<meta-data
    android:name="com.google.firebase.messaging.default_notification_channel_id"
    android:value="rn-push-notification-channel-id-default-4-300" />
```
But it seems like there is an issue with DND which is ignored, I'm looking into it.

**Comment by wbercx** [comment]
> Is it for remote notification ?

@Dallas62 Nah, local notifications. 

It appears to be using the Alarm volume, rather than Notification/Ringtone volumes. If I go to the Clock app and I change the alarm volume in there, that affects the next notification that is displayed.

So far this seems to work, but I haven't done thorough testing yet:

````diff
                AudioAttributes audioAttributes = new AudioAttributes.Builder()
                        .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
-                        .setUsage(AudioAttributes.USAGE_ALARM)
+                        .setUsage(AudioAttributes.USAGE_NOTIFICATION)
                        .build();
````

**Comment by Dallas62** [comment]
@wbercx This change works and solve the DND issue ! 🚀 
I will check 2 others issues and release a fix ASAP.
One problem of this change is it's not applied on installed application, a solution is to change the default vibration (if any like 300 => 301) to force create a new channel, but this workaround is :vomiting_face:

**Comment by wbercx** [comment]
@Dallas62 Nice!

That's one other thing I noticed. It creates two channels now - one with sound and one without - but it uses the same name for both. While some of our users tried to work out why it was so quiet, they were wondering why the same entry appeared twice in the Settings app.

Could we dynamically add ` - silent` at the end of the silent channel's name, or is there a way to address this ourselves in the AndroidManifest?

**Comment by Dallas62** [comment]
@wbercx I can implement a solution to select a name / description based on the ID, and fallback to the default (like the default behaviour).
But this is not the right way to manage this... Since I don't have time to refactor channel management for now, it's probably the best workaround..

By changing:
```
public String getChannelName()
public String getChannelDescription()
```
To
```
public String getChannelName(String channel_id)
public String getChannelDescription(String channel_id)
```
in `RNPushNotificationConfig.java`

**Comment by Dallas62** [comment]
I will release a fix for the current issue, and push on `dev` branch a way to manage different channel name based on channel.

**Comment by Dallas62** [comment]
The version `3.5.2` has been released.
You can test the change for Channel Name on `zo0r/react-native-push-notification#dev`:
```
yarn add zo0r/react-native-push-notification#dev
OR
npm install zo0r/react-native-push-notification#dev
```
I will test it later today, AndroidManifest should be updated like this:
```xml

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                android:value="DEFAULT NOTIFICATION CHANNEL NAME"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                    android:value="DEFAULT NOTIFICATION CHANNEL DESCRIPTION"/>

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name.CHANNEL_ID"
                android:value="YOUR CHANNEL_ID NOTIFICATION CHANNEL NAME"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description.CHANNEL_ID"
                    android:value="YOUR CHANNEL_ID NOTIFICATION CHANNEL DESCRIPTION"/>
```

Default channels id:
- `rn-push-notification-channel-id-default-4-300`
- `rn-push-notification-channel-id-4-300`

**Comment by wbercx** [comment]
@Dallas62 You're a legend, thanks for the quick turnaround on everything! Much appreciated!

**Comment by cristianoccazinsp** [comment]
Just to be sure, we need to use the channels `rn-push-notification-channel-id-default-4-300` and `rn-push-notification-channel-id-4-300`  in our manifest in order to customize the notification description, or can we use our own ids?

**Comment by Dallas62** [comment]
Hi @cristianoccazinsp 
You have to use the ID of the library:
- rn-push-notification-channel-id-default-4-300
- rn-push-notification-channel-id-4-300

Since ID are generated based on local notification parameters (🤮) you can use "custom" channel id if they follow this pattern:
 `rn-push-notification-channel-id(-soundname, default if playSound "-default")-(importance: default "4")-(vibration, default "300")`

**Comment by cristianoccazinsp** [comment]
Thanks. As long as it is documented, I don't think it is so bad.  I haven't been able to test the updates yet, but I will let you know how it goes once I do it.

**Comment by wbercx** [comment]
Apparently even on `3.5.2` after a full reinstall some users are not getting sound when they receive a notification. I cannot reproduce it myself though...

**Comment by cristianoccazinsp** [comment]
@Dallas62  just tried this for local notifications, and it seems to be working fine! 

I didn't have to change the manifest at all, can you clarify if that's required, or just optional if we want custom names for each notification channel?

My current manifest:
```
 <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                android:value="DEFAULT NOTIFICATION CHANNEL NAME"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                    android:value="DEFAULT NOTIFICATION CHANNEL DESCRIPTION"/>
```

**Comment by Dallas62** [comment]
Hi @cristianoccazinsp 
If you use the default entry you mentioned in AndroidManifest, all channels will have the same name.
On `dev` branch, take a look at the Readme I put more information on channels and coming changes

**Comment by d0xi5** [comment]
@Dallas62 Hi, I installed the dev version and on an iPhone XS local notifications only vibrates, no sound. Even for 'default' sound.. iPhone volume is way up high and nothing. Any guess?

**Comment by Dallas62** [comment]
Hi @d0xi5 
This repository in not the iOS one, look at iOS repository.
Regards,

**Comment by d0xi5** [comment]
My bad, posting on IOS issue

**Comment by d0xi5** [comment]
Tried it on Android and Im getting no sound. I have my file in raw directory, and 
`soundName:
        Platform.OS === 'android'
          ? 'android.resource://com.xyz/raw/ding'
          : 'ding.wav'`

Im receiving the notification but its not vibrating nor making any sound. I tried with 'default' and nothing. I have also checked the permissions under the app for sound and they are fine.
Any guess?

**Comment by d0xi5** [comment]
I have added on manifiest the following: 

`

    <uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS"/>

    <uses-permission android:name="android.permission.VIBRATE" />

    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>

  <meta-data
            android:name="com.google.firebase.messaging.default_notification_channel_id"
            android:value="rn-push-notification-channel-id-ding-4-300" />

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
            android:value="Sound channel"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
            android:value="A sound channel"/>

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name.CHANNEL_ID"
            android:value="rn-push-notification-channel-id-ding-4-300"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description.CHANNEL_ID"
            android:value="A sound channel"/>

        
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_foreground"
            android:value="false"/>
        
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
            android:resource="@color/white"/> 

        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationActions" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
                <action android:name="android.intent.action.QUICKBOOT_POWERON" />
                <action android:name="com.htc.intent.action.QUICKBOOT_POWERON"/>
            </intent-filter>
        </receiver>

        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>
`

And on NotifService.js 

`

    PushNotification.createChannel(
      {
        channelId: 'rn-push-notification-channel-id-ding-4-300', // (required)
        channelName: `Sound channel`, // (required)
        channelDescription: 'A sound channel', // (optional) default: undefined.
        soundName:
          Platform.OS === 'android'
            ? 'android.resource://com.ascendapp/raw/ding'
            : 'ding.wav', // (optional) See `soundName` parameter of `localNotification` function
        importance: 4, // (optional) default: 4. Int value of the Android notification importance
        vibrate: true // (optional) default: true. Creates the default vibration patten if true.
      },
      created =>
        console.log(`createChannel 'sound-channel-id' returned '${created}'`) // (optional) callback returns whether the channel was created, false means it already existed.
    )
`

Am I missing something?

**Comment by Dallas62** [comment]
Hi @d0xi5 

Did you try to reinstall the device or change the notification id ?
If you made some test with the same device, the channel might be created and will not be updated (Android Policy).
Also did you try the example project which provide an example with sounds ?

**Comment by d0xi5** [comment]
@Dallas62 I have tried both things and no luck. Just the notification but without sound and without vibration. Phone is on sound mode. 

I couldn't try example app, I had some errors on Android Studio.

**Comment by d0xi5** [comment]
Found the issue, I was doing

`
let notif = new NotifService()
`

Inside the component (hooks), and the channel was being created a lot of times. I passed that code outside of it
and the channel is being created only once, and sound is working. 

Thank you for you quick reply @Dallas62

**Comment by nguyendinhnhan** [comment]
> sound



> > > Check in ur Settings -> Notification -> Your App -> Sound, is a sound chosen?
> > 
> > 
> > Sound is available, but no chosen ("no sound" / "none"). Of course I could choose a sound there, and then it works, but it should be turned on by default.
> 
> Yes, I face the same issue. I believe @Dallas62 is fixing it now

How to make turn on sound by default? Did we fixed it on new version?

**Comment by mitalpatel5944** [comment]
how can we play a custom sound in remote notification ?
i have tried with many solution but it doesn't work on remote notification while custom sound work on local notification,

is there any solution ?

Thanks in advance

**Comment by cihanbey211** [comment]
First, giving 4 to the importance parameter.
The problem will be solved when you remove the playSound and soundName parameters and send the default value.

**Comment by UxmanKaxmi** [comment]
I have a similar problem, If I create a channel and set the soundName as "default" or just remove it, it will always play the default sound, even when I schedule a notification with another custom sound. 
I can however play the custom sound when I add it in the Create Channel Object, I need to play different custom sounds for the scheduled notifications on the same channel? is this possible?

on the latest 7.2.3 version.

**Comment by Dallas62** [comment]
Hi @UxmanKaxmi 
Sounds are defined on channels, you can't override or change the sound of a channel.
Please refer to the Android documentation about channels.
Regards

**Comment by ugar0ff** [comment]
I have same problem on Samsung S9+, Android 10
`"react-native-push-notification": "7.2.3",`

**Comment by TomasPlachy** [comment]
@cristianoccazinsp Hi, could you tell me how did you manage to turn on the sound in the chanell configuration, as you write in the opening post?

My function for scheduling local notifications looks like this (the notification is shown at the right time, but without sound).
`schduleNotification(date) {
        PushNotification.localNotificationSchedule({
            channelId: 'reminders',
            title: '🔔 Reminder!',
            message: 'Reminder ' + date.getMinutes(),
            playSound: true,
            soundName: 'default',
            date,
        });
    }`

**Comment by cristianoccazinsp** [comment]
The code looks fine , what about the channel configuration? Chances are you may need to uninstall and re install the app if the channel was already created as muted.

```
if (Platform.OS == 'android') {
  try {
    PushNotification.createChannel(
      {
        channelId: 'com.myapp', // (required)
        channelName: 'com.myapp', // (required)
        channelDescription: 'Myapp notifications.', // (optional) default: undefined.
        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function

        // (optional) default: 4. Int value of the Android notification importance
        // make it match calls to .localNotification priority
        importance: 4,
        vibrate: true, // (optional) default: true. Creates the default vibration patten if true.
      },
      (created) => {}, // (optional) callback returns whether the channel was created, false means it already existed.
    );
  } catch (err) {
    console.error(
      'Failed to create Android notification channel: ',
      err.message || err,
    );
  }
}
```

**Comment by TomasPlachy** [comment]
@cristianoccazinsp Thank you for replying. I am using AVD and if I understand it correctly it installs the app each time I start up the envirorment. I have tried adding your code to the constructor, but it didnt help (the virtual device plays other sounds from the app, like clicking a button, so I think the problem is not with the emulator).

`
class Notifications {

    constructor() {

        // Must be outside of any component LifeCycle.
        PushNotification.configure({
            onRegister: function (token) {
                console.log("TOKEN:", token);
            },

            // (required) Called when a remote is received or opened, or local notification is opened
            onNotification: function (notification) {
                console.log("NOTIFICATION:", notification);

                // (required) Called when a remote is received or opened, or local notification is opened
                notification.finish(PushNotificationIOS.FetchResult.NoData);
            },

            onAction: function (notification) {
                console.log("ACTION:", notification.action);
                console.log("NOTIFICATION:", notification);
            },

            onRegistrationError: function (err) {
                console.error(err.message, err);
            },

            permissions: {
                alert: true,
                badge: true,
                sound: true,
            },
            popInitialNotification: true,
            requestPermissions: Platform.OS === 'ios',
        });

        if (Platform.OS == 'android') {
            try {
                PushNotification.createChannel(
                    {
                        channelId: 'com.DreamApp2', // (required)
                        channelName: 'com.DreamApp', // (required)
                        channelDescription: 'DreamApp1 notifications.', // (optional) default: undefined.
                        playSound: true,
                        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function

                        // (optional) default: 4. Int value of the Android notification importance
                        // make it match calls to .localNotification priority
                        importance: Importance.HIGH,
                        vibrate: true, // (optional) default: true. Creates the default vibration patten if true.
                    },
                    (created) => { console.log(`createChannel returned '${created}'`) }, // (optional) callback returns whether the channel was created, false means it already existed.
                );
            } catch (err) {
                console.error(
                    'Failed to create Android notification channel: ',
                    err.message || err,
                );
            }
        }

        PushNotification.getScheduledLocalNotifications(rn => {
            console.log('SN --- ', rn);
        });
    }

    schduleNotification(date) {
        PushNotification.localNotificationSchedule({
            channelId: 'com.DreamApp2',
            title: '🔔 Reminder!',
            message: 'Reminder ' + date.getMinutes(),
            playSound: true,
            soundName: 'default',
            date,
        });
    }

}

export default new Notifications();
`

The Notifications class is singleton (instantiated in App.js) which is passed to other components. I then use the scheduleNotification() function to set notifications, which are displayed on time, just without sound (and since the whole point of the app is to be an alarm clock, this is a huge problem for me).

**Comment by cristianoccazinsp** [comment]
Have you tried if regular notifications instead of scheduled ones make any sound?

Also, go to the OS app settings and  check how was the notificación created. 

Lastly, try to test on a real device too, it could be many things.

**Comment by motasimmakki** [comment]
> Gabsys

How can we ask for this permission when the app opens?

**Comment by bibhuti9** [comment]
-> Your channelId and channelName should be same.
now this is work for me

**Comment by R0LLeX** [comment]
The sound in notifications does not work when the app is in the background, can someone help?
`if (Platform.OS === 'android') {
      PushNotification.createChannel(
          {
            channelId: 'channel-id-sound', // (required)
            channelName: 'channelName', // (required)
            channelDescription: 'ChannelName Notifications', // (optional) default: undefined.
            importance: 4,
            soundName: 'boys.mp3',
            vibrate: true,
            playSound: true,
          },
          (created) => __DEV__ && console.log(`createChannel returned '${created}'`),
      )}`

**Comment by sunil-singh-chaudhary** [comment]
> <uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS"/>

<uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS"/>
using this permission play store reject your app in play store i have problem in realme device didnt get notification custom sound working fine in all devices for MI phones i turn on auto restart toggle from setting of device any idea how to do it in realme narzo UI 5.0

<uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS"/>
